coredb-operator/src/controller.rs

```diff
@@ -144,30 +141,15 @@ impl CoreDB {
             .await
             .expect("error reconciling service");
 
-        // Wait for Pod to be running and ready before creating extensions
-        let pod_name = format!("{}-0", name);
-        debug!("Waiting for pod to be running: {}", pod_name);
-        let _check_for_pod = tokio::time::timeout(
-            std::time::Duration::from_secs(timeout_seconds_start_pod),
-            await_condition(pods.clone(), &pod_name, conditions::is_pod_running()),
-        )
-        .await
-        .expect(&format!(
-            "Did not find the pod {} to be running after waiting {} seconds",
-            pod_name, timeout_seconds_start_pod
-        ));
-
-        debug!("Waiting for pod to be ready: {}", pod_name);
-        let _check_for_pod_ready = tokio::time::timeout(
-            std::time::Duration::from_secs(timeout_seconds_pod_ready),
-            await_condition(pods.clone(), &pod_name, is_pod_ready()),
-        )
-        .await
-        .expect(&format!(
-            "Did not find the pod {} to be ready after waiting {} seconds",
-            pod_name, timeout_seconds_pod_ready
-        ));
-        debug!("Found pod ready: {}", pod_name);
+        let primary_pod = self.primary_pod(ctx.client.clone()).await;
+        if primary_pod.is_err() {
+            debug!("Did not find primary pod");
+            return Ok(Action::requeue(Duration::from_secs(1)));
+        }
+        if !is_pod_ready().matches_object(Some(&primary_pod.unwrap())) {
+            debug!("Did not find primary pod is ready");
+            return Ok(Action::requeue(Duration::from_secs(1)));
```

Comment:
This is 1 second wait hardcoded, but I think it ought to be an exponential backoff, up to X seconds maximum backoff time. However, I do not know how to get the previous queue duration yet.

Reply:
https://cloud.redhat.com/blog/kubernetes-operators-best-practices

Reply:
I think they are writing to 'status', that could be a future improvement to prevent checking on stuck pods every second